packages/apollo-angular/testing/src/operation.ts

```diff
@@ -11,21 +13,35 @@ export type Operation = LinkOperation & {
 };
 
 export class TestOperation<T = { [key: string]: any }> {
+  private readonly definition: OperationDefinitionNode | FragmentDefinitionNode;
+
   constructor(
     public readonly operation: Operation,
     private readonly observer: Observer<FetchResult<T>>,
-  ) {}
+  ) {
+    this.definition = getMainDefinition(this.operation.query);
```

Comment:
`this.operation` might be changed later than constructor, so we should use a local variable inside `flush()` instead of `this.definition`